pkg/api/vm/handler.go

```diff
@@ -1164,6 +1181,167 @@ func (h *vmActionHandler) removeVolume(ctx context.Context, namespace, name stri
+	if err != nil {
+		return nil, err
+	}
+	return nad, nil
+}
+
+func isVmNetworkHotpluggable(nad *cniv1.NetworkAttachmentDefinition) (bool, error) {
+	var conf cnitypes.PluginConf
+	if err := json.Unmarshal([]byte(nad.Spec.Config), &conf); err != nil {
+		return false, err
+	}
+
+	if conf.Type != "bridge" {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// addNic add a hotplug NIC with given interface name and network name.
+func (h *vmActionHandler) addNic(ctx context.Context, namespace, name string, input AddNicInput) error {
+	vm, err := h.vmCache.Get(namespace, name)
+	if err != nil {
+		return err
+	}
+
+	nad, err := h.getVmNetwork(input.NetworkName, namespace)
+	if err != nil {
+		return err
+	}
+
+	ok, err := isVmNetworkHotpluggable(nad)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return fmt.Errorf("network %s is not hot-pluggable", input.NetworkName)
+	}
+
+	vmCopy := vm.DeepCopy()
+
+	newIface := kubevirtv1.Interface{
+		Name:                   input.InterfaceName,
+		MacAddress:             input.MacAddress,
+		Model:                  "virtio",
+		InterfaceBindingMethod: kubevirtv1.InterfaceBindingMethod{Bridge: &kubevirtv1.InterfaceBridge{}},
+	}
+	vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces = append(vmCopy.Spec.Template.Spec.Domain.Devices.Interfaces, newIface)
+
+	newNetwork := kubevirtv1.Network{
+		Name:          input.InterfaceName,
+		NetworkSource: kubevirtv1.NetworkSource{Multus: &kubevirtv1.MultusNetwork{NetworkName: input.NetworkName}},
+	}
+	vmCopy.Spec.Template.Spec.Networks = append(vmCopy.Spec.Template.Spec.Networks, newNetwork)
+
+	_, err = h.vms.Update(vmCopy)
+	if err != nil {
+		return err
+	}
+	return h.migrate(ctx, namespace, name, "")
+}
+
+func (h *vmActionHandler) getHotunpluggableNetworks(vm *kubevirtv1.VirtualMachine) (map[string]struct{}, error) {
+	namespace := vm.ObjectMeta.Namespace
+	validNetworks := make(map[string]struct{})
+	for _, network := range vm.Spec.Template.Spec.Networks {
+		if network.Multus != nil {
+			nad, err := h.getVmNetwork(network.Multus.NetworkName, namespace)
+			if err != nil {
+				return nil, err
+			}
+
+			ok, err := isVmNetworkHotpluggable(nad)
+			if err != nil {
+				return nil, err
+			}
+			if ok {
+				validNetworks[network.Name] = struct{}{}
+			}
+		}
+	}
+
+	return validNetworks, nil
+}
+
+func isInterfaceHotunpluggable(iface kubevirtv1.Interface, hotunpluggableNetworks map[string]struct{}) bool {
+	_, isNetworkHotunpluggable := hotunpluggableNetworks[iface.Name]
+	if iface.Model == "virtio" && iface.Bridge != nil && iface.State != kubevirtv1.InterfaceStateAbsent && isNetworkHotunpluggable {
+		return true
+	}
+	return false
+}
+
+// removeNic remove a hotplug NIC by its interface name
+func (h *vmActionHandler) removeNic(ctx context.Context, namespace, name string, input RemoveNicInput) error {
+	vm, err := h.vmCache.Get(namespace, name)
+	if err != nil {
+		return err
+	}
+
+	hotunpluggableNetworks, err := h.getHotunpluggableNetworks(vm)
```

Comment:
during a nic removal we do not really need to check if the underlying network is bridge. we just need to check if we are using the bridge interface or not using a custom binding, and just remove the interface

Reply:
addressed in https://github.com/harvester/harvester/pull/9395/commits/9e42071238bf8e96c534364828e398b03e2b8794